MantidQt/CustomInterfaces/inc/MantidQtCustomInterfaces/Reflectometry/DataProcessorAlgorithm.h

```diff
@@ -0,0 +1,119 @@
+#ifndef MANTID_CUSTOMINTERFACES_DATAPROCESSORALGORITHM_H
+#define MANTID_CUSTOMINTERFACES_DATAPROCESSORALGORITHM_H
+
+#include "MantidAPI/AlgorithmManager.h"
+
+#include <set>
+#include <string>
+#include <vector>
+
+namespace MantidQt {
+namespace CustomInterfaces {
+/** @class DataProcessorAlgorithm
+
+DataProcessorAlgorithm defines a processing algorithm that will perform the
+reduction in a Data Processor UI.
+
+Copyright &copy; 2011-14 ISIS Rutherford Appleton Laboratory, NScD Oak Ridge
+National Laboratory & European Spallation Source
+
+This file is part of Mantid.
+
+Mantid is free software; you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation; either version 3 of the License, or
+(at your option) any later version.
+
+Mantid is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+File change history is stored at: <https://github.com/mantidproject/mantid>.
+Code Documentation is available at: <http://doxygen.mantidproject.org>
+*/
+class DataProcessorAlgorithm {
+public:
+  DataProcessorAlgorithm() = delete;
+  /** Constructor
+  * @param name : The name of this algorithm
+  * @param prefix : The list of prefixes that will be used for the output
+  * workspaces' names
+  * @param blacklist : The list of properties we do not want to show
+  */
+  DataProcessorAlgorithm(
+      const std::string &name, const std::vector<std::string> &prefix,
+      const std::set<std::string> &blacklist = std::set<std::string>())
+      : m_name(name), m_prefix(prefix), m_blacklist(blacklist) {
+
+    Mantid::API::IAlgorithm_sptr alg =
+        Mantid::API::AlgorithmManager::Instance().create(m_name);
+
+    auto properties = alg->getProperties();
+    for (auto &prop : properties) {
+
+      if (prop->direction() == 0 &&
```

Comment:
Yes, they do share code at the moment. In fact that was my impression too and when I first implemented this I did it using a shared base class. But at some point I decided to follow this approach instead to make the three classes more independent, so they would be easier to modify in the future (for instance other people may need their `DataPreprocessorAlgorithm` to allow other type of input properties, `InOut` properties instead of only `Input` or `Output`, etc), right now I don't know how different they may become...